ui/js/components/auth.js

```diff
@@ -0,0 +1,165 @@
+                }
+            });
+
+            // Ask the back-end if a user is logged in on page load
+            ThUserModel.get().then(function (currentUser) {
+                if (currentUser.email) {
+                    ctrl.setLoggedIn(currentUser);
+                } else {
+                    ctrl.setLoggedOut();
+                }
+            });
+
+            /**
+             * Contact login.taskcluster to log the user in.  Opens a new tab
+             * for the tc-login, which will get closed if it's successful.
+             */
+            ctrl.login = function () {
+                var hash = encodeURIComponent("#");
+                var colon = encodeURIComponent(":");
+                var target = `${$location.protocol()}${colon}//${$location.host()}${colon}${$location.port()}/${hash}/login`;
+                var description = "Treeherder";
+                var url = `https://login.taskcluster.net/?target=${target}&description=${description}`;
+
+                // open a new tab to show the taskcluster auth login page
+                $window.open(url, "_blank");
+            };
+
+            /**
+             * Contact Treeherder back-end to log the user out and invalidate
+             * the session token.  Then updates the UI
+             */
+            ctrl.logout = function () {
+                $http.get(thUrl.getRootUrl("/auth/logout/"));
+                ctrl.setLoggedOut();
+            };
+
+            ctrl.setLoggedIn = function(newUser) {
+                newUser.loggedin = true;
+                localStorageService.set("user", newUser);
+                _.extend(ctrl.user, newUser);
+                ctrl.onUserChange({$event: {user: ctrl.user}});
+
+            };
+
+            ctrl.setLoggedOut = function() {
+                localStorageService.set("user", loggedOutUser);
+                _.extend(ctrl.user, loggedOutUser);
+                ctrl.onUserChange({$event: {user: loggedOutUser}});
+            };
+        }]
+});
+
+treeherder.component("loginCallback", {
+    template: `
+        <div ng-if="!loginError">
+            <span>Logging in... <span class="fa fa-spinner fa-spin"></span></span>
+        </div>
+        <div ng-if="loginError">
+            <span>Error logging in: {{loginError}}</span>
+        </div>
+    `,
+    controller: ['localStorageService', '$location', '$window', '$http', '$scope',
+        function(localStorageService, $location, $window, $http, $scope) {
+            const host = $location.host();
+            const port = $location.port();
+            const urlBase = `${$location.protocol()}://${host}:${port}/`;
+            const certificate = $location.search().certificate;
+            const credentials = {
+                id: $location.search().clientId,
+                key: $location.search().accessToken,
+                algorithm: 'sha256'
+            };
+            this.loginError = null;
+            const header = hawk.client.header(urlBase, 'GET', {
```

Comment:
Should we be Hawk signing `urlBase` or the full login URL? I think the latter seems more accurate? ie: The backend would then use a `resource` of "/api/auth/login/" not "/".